atox/src/main/kotlin/ui/user_profile/UserProfileFragment.kt

```diff
@@ -155,23 +170,101 @@ class UserProfileFragment : BaseFragment<FragmentUserProfileBinding>(FragmentUse
+        val bmp = createQrCode()
         val qrCode = ImageView(requireContext()).apply {
-            setPadding(dpToPx(QR_CODE_DIALOG_PADDING, resources))
+            setPadding(qrCodePadding.asPx(resources).px)
             setImageBitmap(bmp)
         }
+
         return AlertDialog.Builder(requireContext())
             .setTitle(R.string.tox_id)
             .setView(qrCode)
+            .setPositiveButton(getString(R.string.share)) { _, _ ->
+                vm.viewModelScope.launch {
+                    val qrImageUri = getQrForSharing("tox_id_qr_code")
+                    val shareIntent = Intent().apply {
+                        action = Intent.ACTION_SEND
+                        clipData = ClipData.newRawUri(null, qrImageUri)
+                        type = "image/png"
+                        putExtra(Intent.EXTRA_STREAM, qrImageUri)
+                        flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+                    }
+                    startActivity(Intent.createChooser(shareIntent, getString(R.string.tox_id_share)))
+                }
+            }
+            .setNegativeButton(getString(R.string.close), null)
             .create()
     }
+
+    /*
+     * Function will save the image with the input bitmap and name to the cache directory as a png format.
+     * Then it will return the Uri of the image.
+     */
+    private fun saveImageForSharing(context: Context, image: Bitmap, name: String): Uri {
+        val imagesFolder = File(context.cacheDir, "shared_images")
+        val uri: Uri
+
+        imagesFolder.mkdirs()
+        val file = File(imagesFolder, "$name.png")
+        FileOutputStream(file).use { stream ->
+            image.compress(Bitmap.CompressFormat.PNG, 90, stream)
+            stream.flush()
+        }
+        uri = FileProvider.getUriForFile(context, "${BuildConfig.APPLICATION_ID}.fileprovider", file)
+
+        return uri
+    }
+
+    // Function will create and return the QR code bitmap of the Tox ID.
+    private fun createQrCode(
+        qrSize: Size = Px(
+            (
+                min(
+                    resources.displayMetrics.widthPixels,
+                    resources.displayMetrics.heightPixels
+                ) * QR_CODE_TO_SCREEN_RATIO
+                ).toInt()
+        ),
+        padding: Size = qrCodePadding
+    ): Bitmap {
+        // Creating the QR bitmap
+        val qrData = QrCode.encodeText("tox:%s".format(vm.toxId.string()), QrCode.Ecc.LOW)
+        val qrCodeColor: Int = Color.BLACK
+        val backgroundColor: Int = Color.WHITE
+        var bmpQr: Bitmap = Bitmap.createBitmap(qrData.size, qrData.size, Bitmap.Config.ARGB_8888)
+        bmpQr.setHasAlpha(true)
+        for (x in 0 until qrData.size) {
+            for (y in 0 until qrData.size) {
+                bmpQr.setPixel(x, y, if (qrData.getModule(x, y)) qrCodeColor else backgroundColor)
+            }
+        }
+        // Scaling the QR bitmap to be half of the screen's width
```

Comment:
Drop the comments in this function. In general, only comment on things that aren't obvious when looking at the code. E.g. this comment is obvious since you're calling `scale` on the `bmpQr` variable. That it says "half of the screen's width" won't always be correct since the size is passed as an argument.

Reply:
Oops. An old comment before I added `qrSize`.

Reply:
Heh, that's the exact reason I don't want comments to say things that are obvious from looking at the code. :P Same info, but twice, so you have to keep it in sync.